### PR TITLE
added in the 07 api end point

### PR DESCRIPTION
**Pull request by vaishkhullar**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -133,3 +133,52 @@ describe("/api/articles/:article_id/comments", () => {
...
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("bad request");
+      });
+  });
+});
```
Comment (Jodiech) [inline __tests__/app.test.js:212]: You need a test for a status 400, missing required field(s), e.g. no username or body properties

File: __tests__/app.test.js [hunk 2]
```diff
@@ -133,3 +133,52 @@ describe("/api/articles/:article_id/comments", () => {
...
+
+  test("should return a sql error if the username doesn't exist in the users table ", () => {
+    const newComment = {
+      username: "vaish",
+      body: "best comment ever.",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
```
Comment (Jodiech) [inline __tests__/app.test.js:207]: This should be a status 404, username does not exist.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -133,3 +133,52 @@ describe("/api/articles/:article_id/comments", () => {
...
+      .then(({ body: { comment } }) =>
+        expect.objectContaining({
+          author: expect(newComment.username),
+          body: expect(newComment.body),
+          votes: expect(0),
+          article_id: expect(1),
+        })
+      );
+  });
+
```
Comment (Jodiech) [inline __tests__/app.test.js:171]: You need a status 201, ignores unnecessary properties for example -    
 const newComment = {
      username: "butter_bridge",
      body: "best comment ever.",
      banana: "fruit"
    };

**Review by Jodiech — APPROVED** [review]
Well done Vaish! I've left a few comments for you to look at and amend. Please add some more tests as per the comments. Happy for you to merge when this has been actioned.

**Comment by vaishkhullar** [comment]
Ace, I’ll go back and amend once I’ve finished 09 ticket - thanks Jodie ☺️

Sent from my iPhone

On 16 Dec 2022, at 09:34, Jodie C ***@***.***> wrote:

﻿

@Jodiech approved this pull request.

Well done Vaish! I've left a few comments for you to look at and amend. Please add some more tests as per the comments. Happy for you to merge when this has been actioned.

________________________________

In __tests__/app.test.js<https://github.com/vaishkhullar/openAPI/pull/2#discussion_r1050536277>:

> +  });
+
+  test("should return a sql error if the username doesn't exist in the users table ", () => {
+    const newComment = {
+      username: "vaish",
+      body: "best comment ever.",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("bad request");
+      });
+  });
+});


You need a test for a status 400, missing required field(s), e.g. no username or body properties

________________________________

In __tests__/app.test.js<https://github.com/vaishkhullar/openAPI/pull/2#discussion_r1050536784>:

> +      .send(newComment)
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("not found");
+      });
+  });
+
+  test("should return a sql error if the username doesn't exist in the users table ", () => {
+    const newComment = {
+      username: "vaish",
+      body: "best comment ever.",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(400)


This should be a status 404, username does not exist.

________________________________

In __tests__/app.test.js<https://github.com/vaishkhullar/openAPI/pull/2#discussion_r1050538335>:

> +      body: "best comment ever.",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body: { comment } }) =>
+        expect.objectContaining({
+          author: expect(newComment.username),
+          body: expect(newComment.body),
+          votes: expect(0),
+          article_id: expect(1),
+        })
+      );
+  });
+


You need a status 201, ignores unnecessary properties for example -
const newComment = {
username: "butter_bridge",
body: "best comment ever.",
banana: "fruit"
};

**Pull request merged**